src/service/crypto-impl.js

```diff
@@ -213,6 +213,41 @@ export class Crypto {
       data
     ));
   }
+
+  /**
+   * Returns a cryptographically random string with 128 bits of entropy
+   * to be used as a subdomain for SafeFrame.
+   * @return {string}
+   */
+  getSecureRandomString() {
+    // Support IE 11
+    const cryptoLib =
+      /** @type {!webCrypto.Crypto|undefined} */ (this.win_.crypto ||
+      this.win_.msCrypto);
+    let /** !Array<number>|!Uint8Array */ randomValues;
+
+    if (cryptoLib !== undefined) {
+      randomValues = new Uint8Array(16);
```

Comment:
I suggest allowing the array size to be passed as an argument and returning cryptoLib.getRandomValues(randomValues) when cryptoLib is available and null otherwise.  Then within amp-a4a.js, put your polyfill handling null case.

Reply:
Still not convinced we should put fallback logic here esp given comments suggest you're returning a "cryptographically" random string when you're not if crypto is not available.

Reply:
That makes sense. I'll change it

Reply:
Working on changing it. Perhaps we could rename this function instead, to something like `getSecureRandomStringWithFallback` ?

Reply:
Actually, if we want to split this function across two different functions some logic would need to be duplicated. Perhaps it would be better to just keep it here and change the comments/params? I just added a param (by default is off) to use the fallback when win.crypto isn't available.

Reply:
Yeah I still think the fallback code should only be within doubleclick impl for now where it should return null if no crypto available.

Reply:
done